base/path.jl

```diff
@@ -248,16 +248,17 @@ function splitpath(p::String)
     return out
 end
 
-joinpath(path::AbstractString)::String = path
-
 if Sys.iswindows()
 
-function joinpath(path::AbstractString, paths::AbstractString...)::String
-    result_drive, result_path = splitdrive(path)
+function joinpath(paths::Union{Tuple, AbstractVector{<:AbstractString}})::String
```

Comment:
```suggestion
function joinpath(paths::Union{Tuple, AbstractVector})::String
```
The real benefit of not being too strict with the types here is that someone can pass an `Any` vector that happens to only contain strings and it will work. The same applies to the other signatures.